### ISSUE TITLE
LibreOffice Writer: in Confirm dialog in Document restore is not announced when using screenreaders

### ISSUE THREAD

**Issue by peterpacult** [nvaccess/nvda#11687]
### Steps to reproduce:
1. Crash LibreOffice
2. restart LibreOffice
3. Dissmiss restoring Documents
4. The confirm dialog opens and all you can here from screenreader is "Yes" or "No".

### Actual behavior:
Only the buttons (Yes or No) are announced by screenreader.

### Expected behavior:
The question, which this dialog asks should be announced.

#### NVDA installed/portable/running from source:
installed
#### NVDA version:
2020.2

#### Windows version:
Edition	Windows 	10 Home
Version				1909
Operating system build	18363.1016

#### Name and version of other software in use when reproducing the issue:
LibreOffice
Version:	7.0.0.3 (x64)
Build ID:	8061b3e9204bef6b321a21033174034a5e2ea88e
CPU-Threads: 	4; BS: Windows 10.0 Build 18363; UI-Render: Skia/Vulkan; VCL: win
Locale: 	de-DE (de_DE); UI: de-DE
Calc: 	threaded

#### Other information about your system:
---
#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.

#### If addons are disabled, is your problem still occuring?
Not tested.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
Not tested.

**Comment by peterpacult** [comment]
This issue has also been reported to the Document Foundation; see https://bugs.documentfoundation.org/show_bug.cgi?id=135588

**Comment by michaelweghorn** [comment]
I have made two related changes in LibreOffice, s. https://bugs.documentfoundation.org/show_bug.cgi?id=135588 .
An additional change is needed to make sure that the dialog text is actually announced. From how I understand it, that is best handled on NVDA side, and I have created a pull request for this:  #12801

I'd be happy to receive feedback on this.

**Comment by peterpacult** [comment]
Hello Michael,

I have read through your pull request message. It all sounds very
promising. However, I don't currently have LibreOffice version 7.3 on the
computer, nor an alpha version of NVDA 21.2, so unfortunately I can't test
that myself at the moment. I am just a "simple" NVDA and LibreOffice user
and not a developer...
To be able to support you further, I would need more information -
currently my level of knowledge is not sufficient to even know what exactly
I should actually do...

With best regards

Am Do., 2. Sept. 2021 um 13:41 Uhr schrieb Michael Weghorn <
***@***.***>:

**Comment by michaelweghorn** [comment]
Hello Peter,
thanks for your readiness to help here. This is the first time I am trying to contribute to NVDA, so many things are new to me as well. My request for feedback to my pull request was primarily targeted at NVDA developers. 

If you wanted to test this as a user, you would need LibreOffice 7.3 (i.e. the current development version), as well as NVDA with the change from my pull request included. (It is not contained in any alpha version or official development snapshot as of now.)

If you want, you can download a current development version of LibreOffice 7.3 from here: 
https://dev-builds.libreoffice.org/daily/master/Win-x86_64@tb77-TDF/current/
That page links to the MSI installer files. Unless you want to have support for additional languages or want to install the help files, you'll just need the first MSI installer from there, called LibreOfficeDev_7.3.0.0.alpha0_Win_x64.msi . If you install this, it will install the LibreOffice development version as "LibreOfficeDev", and will keep your stable LibreOffice version unchanged. (That means you can choose whether to start the stable version of LibreOffice or the development version.)

However, I don't know whether there's an easy way to get an NVDA version including the change from my pull request. (An automated test build is done for the pull request, but I do not know whether the resulting NVDA build artifact is offered for download anywhere. (Maybe people more experienced with NVDA's CI setup can help here...)

**Comment by CyrilleB79** [comment]
Here is a [download link](https://ci.appveyor.com/api/buildjobs/2e8rkafw34egi93j/artifacts/output%2Fnvda_snapshot_pr12801-23670%2C0490c8bd.exe) for the build generated by the PR #12801 .

FYI, to find it from the PR, just click on the details link of the appVeyor paragraph and you will get to the appVeyor page of this build. Click the artifact tab and you will find all the artifacts generated for this commit, among which the snapshot build.

**Comment by michaelweghorn** [comment]
@CyrilleB79: Thanks!
